### ISSUE TITLE
[Impeller] AHB swapchains broken on Samsung Galaxy Note 10.

### ISSUE THREAD

**Issue by benedicte69** [flutter/flutter#160119]
### Steps to reproduce

Create a Flutter app.
Set up an Android Studio Emulator with Vulkan support enabled.
Set up a physical Android device (e.g., Android-arm64, Android 13, API 33).
Run the app with Impeller enabled (default).
Observe that the app crashes after a short time on the emulator and on the physical device it is an absolute blackscreen
Disable Impeller by running the app with the --no-impeller flag.
Observe that the app runs correctly without crashes on both the emulator and physical device.

### Expected results

The app should run without crashes on both Android Studio Emulators and physical Android devices, regardless of whether Impeller or Skia is used. The app should work seamlessly with Vulkan support (using Impeller) and without it (using Skia).

### Actual results

The app crashes after a short time when Impeller is enabled (by default) on Android Studio Emulators e.g. pixel api 35 vanilla 5.0 1080x1920

The app is rendering a full black screen on real device (android-arm64  • Android 13 (API 33)
), I can see only the number 63(which is the number of notifications received), nothing else

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[flutter run --no-enable-impeller  SAVE MY LIFE !]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Android Studio Ladybug | 2024.2.1 Patch 3
Build #AI-242.23339.11.2421.12700392, built on November 22, 2024
Runtime version: 21.0.3+-12282718-b509.11 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.awt.windows.WToolkit
Windows 11.0
Kotlin plugin: K2 mode (Beta)
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 4058M
Cores: 8
Registry:
  debugger.valueTooltipAutoShowOnSelection=true
  ide.experimental.ui=true
  i18n.locale=
Non-Bundled Plugins:
  Dart (242.24931)
  io.flutter (83.0.3)
]
```

</details>

**Comment by jonahwilliams** [comment]
> Set up a physical Android device (e.g., Android-arm64, Android 13, API 33).

What physical device model are you using?

**Comment by WhistlerChoi** [comment]
After updating to 3.27, an error occurs.

**Comment by benedicte69** [comment]
@Jonah Williams => Samsung galaxy note 10, Android 13  One UI 5.1

Yahoo Mail : Recherchez, organisez, maîtrisez 
 
  Le jeu., déc. 12, 2024 à 5:14, ***@***.***> a écrit:   


After updating to 3.27, an error occurs.

**Comment by jonahwilliams** [comment]
https://www.gsmarena.com/samsung_galaxy_note10-9788.php

Do you know if you have the Mali-G76 MP12 - EMEA/LATAM or the Adreno 640 - USA/China ?

**Comment by benedicte69** [comment]
apparently mine is: SAOMC_SM-N770F_OXM_XEF_13_0010 XEF/XEF,XEF/XEF
    Le vendredi 13 décembre 2024 à 00:37:26 UTC+1, Jonah Williams ***@***.***> a écrit :  
 
 


https://www.gsmarena.com/samsung_galaxy_note10-9788.php

Do you know if you have the Mali-G76 MP12 - EMEA/LATAM or the Adreno 640 - USA/China ?

**Comment by benedicte69** [comment]
meaning a french one: 
France
AUC, BOG, COR, DIX, FTM, NRJ, OFR, ORC, ORF, OXA, SFR, UNI, VGF, XEF
    Le vendredi 13 décembre 2024 à 01:02:52 UTC+1, Benedicte ***@***.***> a écrit :  
 
 

 apparently mine is: SAOMC_SM-N770F_OXM_XEF_13_0010 XEF/XEF,XEF/XEF
 Le vendredi 13 décembre 2024 à 00:37:26 UTC+1, Jonah Williams ***@***.***> a écrit : 
 
 


https://www.gsmarena.com/samsung_galaxy_note10-9788.php

Do you know if you have the Mali-G76 MP12 - EMEA/LATAM or the Adreno 640 - USA/China ?

**Comment by jonahwilliams** [comment]
kk, one quick thing to try, Can you add the following line to your AndroidManifest.xml:

```
<meta-data
    android:name="io.flutter.embedding.android.DisableSurfaceControl"
    android:value="true" />
```

**Comment by benedicte69** [comment]
1-create a flutter app2-add to the manifest (c.f: kk.txt in P.J.):         <meta-data            android:name="io.flutter.embedding.android.DisableSurfaceControl"            android:value="true" />3-run the app on a samsung device
Well done! It worksthere is no black screen! I can push the floating action button of Flutter Demo Home Page. 


I read you tomorrow. I go to have sweet dreams.
N.B.: what do you mean by KK ? because in french 'kk' is pronouced 'KaKa'  like 'caca' which means, well I let you translate... I guess you meant Coucou! 

    Le vendredi 13 décembre 2024 à 01:26:14 UTC+1, Jonah Williams ***@***.***> a écrit :  
 
 


kk, one quick thing to try, Can you add the following line to your AndroidManifest.xml:
<meta-data
    android:name="io.flutter.embedding.android.DisableSurfaceControl"
    android:value="true" />

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>
  
<manifest xmlns:android="http://schemas.android.com/apk/res/android">
    <application
        android:label="kk"
        android:name="${applicationName}"
        ***@***.***/ic_launcher">
        <meta-data
            android:name="io.flutter.embedding.android.DisableSurfaceControl"
            android:value="true" />
        <activity
            android:name=".MainActivity"
            android:exported="true"
            android:launchMode="singleTop"
            android:taskAffinity=""
            ***@***.***/LaunchTheme"
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:hardwareAccelerated="true"
            android:windowSoftInputMode="adjustResize">
            
            <meta-data
              android:name="io.flutter.embedding.android.NormalTheme"
              ***@***.***/NormalTheme"
              />
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
        </activity>
        
        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
    </application>
    
    <queries>
        <intent>
            <action android:name="android.intent.action.PROCESS_TEXT"/>
            <data android:mimeType="text/plain"/>
        </intent>
    </queries>
</manifest>

**Comment by darshankawar** [comment]
@benedicte69 Please try to avoid replying via email to the comments, as it doesn't allow us to format the comments and then it becomes difficult to read what you have posted.

**Comment by benedicte69** [comment]
@darshankawar the preview on edge chronium is not working as expected. I redo it in Chrome.
Here the last mail with few more comments:

- 1-create a flutter app
- 2-add to the manifest (c.f: kk.txt in P.J.):         
```
<meta-data           
android:name="io.flutter.embedding.android.DisableSurfaceControl"           
android:value="true" />
```
- 3-run the app on a samsung device

no black screen! I can push the floating action button of Flutter Demo Home Page. 

```
<manifest xmlns:android="http://schemas.android.com/apk/res/android">
    <application
        android:label="kk"
        android:name="${applicationName}"
        android:icon="@mipmap/ic_launcher">
        <meta-data
            android:name="io.flutter.embedding.android.DisableSurfaceControl"
            android:value="true" />
        <activity
            android:name=".MainActivity"
            android:exported="true"
            android:launchMode="singleTop"
            android:taskAffinity=""
            android:theme="@style/LaunchTheme"
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:hardwareAccelerated="true"
            android:windowSoftInputMode="adjustResize">
            <!-- Specifies an Android theme to apply to this Activity as soon as
                 the Android process has started. This theme is visible to the user
                 while the Flutter UI initializes. After that, this theme continues
                 to determine the Window background behind the Flutter UI. -->
            <meta-data
              android:name="io.flutter.embedding.android.NormalTheme"
              android:resource="@style/NormalTheme"
              />
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
        </activity>
        <!-- Don't delete the meta-data below.
             This is used by the Flutter tool to generate GeneratedPluginRegistrant.java -->
        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
    </application>
    <!-- Required to query activities that can process text, see:
         https://developer.android.com/training/package-visibility and
         https://developer.android.com/reference/android/content/Intent#ACTION_PROCESS_TEXT.

         In particular, this is used by the Flutter engine in io.flutter.plugin.text.ProcessTextPlugin. -->
    <queries>
        <intent>
            <action android:name="android.intent.action.PROCESS_TEXT"/>
            <data android:mimeType="text/plain"/>
        </intent>
    </queries>
</manifest>

```
@jonahwilliams By focusing first on Impeller (--no-enable-impeller), I can maintain as much modern rendering functionality as possible, while disabling SurfaceControl is an additional safeguard for device-specific compatibility.

--no-enable-impeller is the preferred solution because it retains modern Android rendering optimizations like SurfaceControl, isn't it ? Then I can revisit Impeller compatibility in future Android Studio Emulator updates, as Impeller will eventually replace Skia ?

I don't remember on which device it was but I tested my app on the remote samsung lab and it was not crashing. So I think the incompatibility is more related to impeller

@darshankawar does this post suit your format ?

**Comment by benedicte69** [comment]
then the reality is that I don't understand everything you're talking about, e.g. 'kk' -that's humour' or more exactly I mean teck terms

**Comment by jonahwilliams** [comment]
>--no-enable-impeller is the preferred solution because it retains modern Android rendering optimizations like SurfaceControl, isn't it ? Then I can revisit Impeller compatibility in future Android Studio Emulator updates, as Impeller will eventually replace Skia ?

SurfaceControl is never used with the Skia rendering backend. Disabling surface control is preferred.

We believe this will be fixed by [#57173
](https://github.com/flutter/engine/pull/57173) which disables surface control for all devices.

**Comment by jonahwilliams** [comment]
Flutter 3.27.1 is available on the stable channel and should have the fix for the originally reported issue.
